src/main/java/org/elasticsearch/cloud/azure/storage/AzureStorageServiceImpl.java

```diff
@@ -79,13 +79,15 @@ public AzureStorageServiceImpl(Settings settings, SettingsFilter settingsFilter)
                 // Create the blob client.
                 client = storageAccount.createCloudBlobClient();
 
-                // Set timeout option. Defaults to 5mn. See cloud.azure.storage.timeout or cloud.azure.storage.xxx.timeout
-                try {
-                    int timeoutAsInt = (int) timeout.getMillis();
-                    client.getDefaultRequestOptions().setMaximumExecutionTimeInMs(timeoutAsInt);
-                } catch (ClassCastException e) {
-                    throw new IllegalArgumentException("Can not convert [" + timeout +
-                            "]. It can not be longer than 2,147,483,647ms.");
+                // Set timeout option. See cloud.azure.storage.timeout or cloud.azure.storage.xxx.timeout
+                if (timeout.getSeconds() > 0) {
+                    try {
+                        int timeoutAsInt = (int) timeout.getMillis();
+                        client.getDefaultRequestOptions().setTimeoutIntervalInMs(timeoutAsInt);
+                    } catch (ClassCastException e) {
```

Comment:
The cast won't cause the exception that you want. JDK 8 added [`Math.toIntExact`](http://docs.oracle.com/javase/8/docs/api/java/lang/Math.html#toIntExact-long-), but since this is for Java 7+, you can't use it and you'll need a helper method to get this effect.